tests/kernel/blktests.pm

```diff
@@ -0,0 +1,78 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018-2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Block device layer tests
+# Maintainer: Michael Moese <mmoese@suse.de>, Sebastian Chlad <schlad@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+#use utils 'zypper_call';
+#use power_action_utils 'power_action';
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    #below variable exposes blktests options to the openQA testsuite
+    #definition, so that it allows flexible ways of re-runing the tests
+    my $quick   = get_required_var('blk_QUICK');
+    my $exclude = get_required_var('blk_EXCLUDE');
+    my $config  = get_required_var('blk_CONFIG');
+    my $tests   = get_required_var('blk_TESTS');
+    my $device  = get_required_var('blk_DEVICE_ONLY');
+
+    script_run('zypper -n in git-core make gcc coreutils awk util-linux fio gcc-c++',       180);
+    script_run('git -c http.sslVerify=false clone https://github.com/osandov/blktests.git', 120);
+
+    script_run('cd ./blktests');
+    script_run('make && make install');
+
+    script_run('./check loop',                 180);
```

Comment:
Again, I would suggest we run one group of blktests at a time, and have multiple testsuites. We can configure them to our liking with hardware we need (like NVMe) and get shorter single job runtimes.

Reply:
wouldn't be better to use blktests as package? it is already in factory in obs/ibs, and it can be easily linked to qa head repo

Reply:
as in the PR message: "(1): I use sourceode and do not package blktests, as I am not sure if this is worth the effort atm. We might do this later, if there is obvious benefit"

Reply:
I copied blktests to QA:Head and you just need to add repo, see  https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/c7c4bff2e4fd03e3997d4e8282ff777a0600b8aa/tests/kernel/install_ltp.pm#L231

Effort should be minimal.

Reply:
who is the maintainer of that rpm btw?

Reply:
According to changes file, it is @frankenmichl :)

Reply:
so I wonder how often does he (or any other maintainer of rpms with tests only) updates the rpm. So I don't really argue to not use rpm. I would start going git clone. See how beneficial will be those tests and only then wonder about packaging and maintaining such package.

It if would be heavy code requiring compiling, then the benefit of starting with rpm is obvious. But given we talk here about ,sh scripts and small thing to be compiled I would still stick to the proposed plan -> start with git clone. See if those tests are useful. If they are useful -> start maintaining the rpm.

Reply:
I created the package when I was playing with this. I haven't updated it in a while, but it should not be an issue doing that.